### ISSUE TITLE
DMF export is broken

### ISSUE THREAD

**Issue by mithkr** [ShadelessFox/decima#53]
DMF export is broken after 0.1.9 (last working release). I have bisect with git and found that export is broken since commit
c81ef53989c389154849a847b49d60fc238c3011

```
~/Development/decima> git bisect reset bisect/bad
Vorherige Position von HEAD war e616f19 Core Editor: Add shortcut for "Inspect Value"
HEAD ist jetzt bei c81ef53 RTTI: Improve data source reading
~/Development/decima> gradlew build

BUILD SUCCESSFUL in 1s
9 actionable tasks: 7 executed, 2 up-to-date
~/Development/decima> tar -xf build/distributions/decima-0.1.9.tar 
~/Development/decima> ./decima-0.1.9/bin/decima
20:28:45.470 [main] INFO org.reflections.Reflections - Reflections took 114 ms to scan 1 urls, producing 192 keys and 777 values
20:28:46.439 [SwingWorker-pool-1-thread-3] DEBUG com.shade.decima.model.rtti.registry.providers.ExternalTypeProvider - Found message handler for type 'DataBufferResource' that handles message 'MsgReadBinary'
20:28:46.444 [SwingWorker-pool-1-thread-3] DEBUG com.shade.decima.model.rtti.registry.providers.ExternalTypeProvider - Found message handler for type 'LocalizedTextResource' that handles message 'MsgReadBinary'
20:28:46.446 [SwingWorker-pool-1-thread-3] DEBUG com.shade.decima.model.rtti.registry.providers.ExternalTypeProvider - Found message handler for type 'PhysicsShapeResource' that handles message 'MsgReadBinary'
20:28:46.448 [SwingWorker-pool-1-thread-3] DEBUG com.shade.decima.model.rtti.registry.providers.ExternalTypeProvider - Found message handler for type 'WaveResource' that handles message 'MsgReadBinary'
20:28:46.449 [SwingWorker-pool-1-thread-3] DEBUG com.shade.decima.model.rtti.registry.providers.ExternalTypeProvider - Found message handler for type 'IndexArrayResource' that handles message 'MsgReadBinary'
20:28:46.450 [SwingWorker-pool-1-thread-3] DEBUG com.shade.decima.model.rtti.registry.providers.ExternalTypeProvider - Found message handler for type 'TextureList' that handles message 'MsgReadBinary'
20:28:46.451 [SwingWorker-pool-1-thread-3] DEBUG com.shade.decima.model.rtti.registry.providers.ExternalTypeProvider - Found message handler for type 'Texture' that handles message 'MsgReadBinary'
20:28:46.456 [SwingWorker-pool-1-thread-3] DEBUG com.shade.decima.model.rtti.registry.providers.ExternalTypeProvider - Found message handler for type 'VertexArrayResource' that handles message 'MsgReadBinary'
20:28:46.458 [SwingWorker-pool-1-thread-3] DEBUG com.shade.decima.model.rtti.registry.providers.ExternalTypeProvider - Found message handler for type 'UITexture' that handles message 'MsgReadBinary'
20:28:46.460 [SwingWorker-pool-1-thread-3] DEBUG com.shade.decima.model.rtti.registry.providers.ExternalTypeProvider - Found message handler for type 'Pose' that handles message 'MsgReadBinary'
20:28:46.930 [ForkJoinPool.commonPool-worker-1] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/DLC1_Arabic.bin'
20:28:46.938 [ForkJoinPool.commonPool-worker-3] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/DLC1_Italian.bin'
20:28:46.950 [ForkJoinPool.commonPool-worker-3] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/DLC1_German.bin'
20:28:46.961 [ForkJoinPool.commonPool-worker-3] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/DLC1_Russian.bin'
20:28:46.962 [ForkJoinPool.commonPool-worker-5] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Initial_LATAMSP.bin'
20:28:46.964 [ForkJoinPool.commonPool-worker-4] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Initial_German.bin'
20:28:46.971 [ForkJoinPool.commonPool-worker-3] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/DLC1_Polish.bin'
20:28:46.982 [ForkJoinPool.commonPool-worker-3] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/DLC1_Portugese.bin'
20:28:46.983 [ForkJoinPool.commonPool-worker-2] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Remainder_LATAMPOR.bin'
20:28:46.986 [ForkJoinPool.commonPool-worker-1] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Initial_English.bin'
20:28:46.992 [ForkJoinPool.commonPool-worker-5] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Initial_LATAMPOR.bin'
20:28:46.992 [ForkJoinPool.commonPool-worker-3] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/DLC1_French.bin'
20:28:47.003 [ForkJoinPool.commonPool-worker-3] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/DLC1_Spanish.bin'
20:28:47.013 [ForkJoinPool.commonPool-worker-3] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/DLC1_English.bin'
20:28:47.017 [ForkJoinPool.commonPool-worker-5] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Initial_Polish.bin'
20:28:47.017 [ForkJoinPool.commonPool-worker-4] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Initial_Portugese.bin'
20:28:47.047 [ForkJoinPool.commonPool-worker-2] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Remainder_LATAMSP.bin'
20:28:47.052 [ForkJoinPool.commonPool-worker-5] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Initial_Russian.bin'
20:28:47.054 [ForkJoinPool.commonPool-worker-2] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/FGRWin32.bin'
20:28:47.066 [ForkJoinPool.commonPool-worker-4] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Initial_Italian.bin'
20:28:47.087 [ForkJoinPool.commonPool-worker-5] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Remainder_French.bin'
20:28:47.103 [ForkJoinPool.commonPool-worker-4] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Initial_Spanish.bin'
20:28:47.110 [ForkJoinPool.commonPool-worker-5] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Remainder_Spanish.bin'
20:28:47.120 [ForkJoinPool.commonPool-worker-5] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Remainder_English.bin'
20:28:47.122 [ForkJoinPool.commonPool-worker-4] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Initial_French.bin'
20:28:47.124 [ForkJoinPool.commonPool-worker-4] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/DLC1_LATAMPOR.bin'
20:28:47.126 [ForkJoinPool.commonPool-worker-5] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Initial_Arabic.bin'
20:28:47.131 [ForkJoinPool.commonPool-worker-4] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/DLC1_LATAMSP.bin'
20:28:47.135 [ForkJoinPool.commonPool-worker-5] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Remainder_Portugese.bin'
20:28:47.141 [ForkJoinPool.commonPool-worker-5] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Remainder_Polish.bin'
20:28:47.145 [ForkJoinPool.commonPool-worker-4] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Remainder_Italian.bin'
20:28:47.154 [ForkJoinPool.commonPool-worker-5] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Remainder_Russian.bin'
20:28:47.155 [ForkJoinPool.commonPool-worker-4] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Remainder_German.bin'
20:28:47.164 [ForkJoinPool.commonPool-worker-5] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Remainder_Arabic.bin'
20:28:47.269 [ForkJoinPool.commonPool-worker-3] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/DLC1.bin'
20:28:47.281 [ForkJoinPool.commonPool-worker-2] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Patch.bin'
20:28:47.374 [ForkJoinPool.commonPool-worker-1] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Initial.bin'
20:28:47.459 [SwingWorker-pool-1-thread-3] INFO com.shade.decima.model.packfile.PackfileManager - Mounted '/data/SteamLibrary/steamapps/common/Horizon Zero Dawn/Packed_DX12/Remainder.bin'
20:29:03.861 [SwingWorker-pool-1-thread-3] INFO com.shade.decima.ui.data.viewer.model.DMFExporter - Exporting RegularSkinnedMeshResource
java.lang.RuntimeException: java.lang.IndexOutOfBoundsException
        at com.shade.decima.ui.data.viewer.model.ModelViewerPanel.lambda$new$1(ModelViewerPanel.java:79)
        at com.shade.platform.ui.dialogs.ProgressDialog$1.doInBackground(ProgressDialog.java:37)
        at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IndexOutOfBoundsException
        at java.base/java.util.zip.DeflaterOutputStream.write(DeflaterOutputStream.java:203)
        at java.base/java.io.ByteArrayInputStream.transferTo(ByteArrayInputStream.java:208)
        at com.shade.decima.ui.data.viewer.model.dmf.DMFInternalBuffer.serialize(DMFInternalBuffer.java:26)
        at com.shade.decima.ui.data.viewer.model.DMFExporter$JsonBufferSerializer.serialize(DMFExporter.java:1515)
        at com.shade.decima.ui.data.viewer.model.DMFExporter$JsonBufferSerializer.serialize(DMFExporter.java:1511)
        at com.google.gson.internal.bind.TreeTypeAdapter.write(TreeTypeAdapter.java:88)
        at com.google.gson.Gson.toJson(Gson.java:825)
        at com.google.gson.Gson.toJsonTree(Gson.java:695)
        at com.google.gson.Gson.toJsonTree(Gson.java:672)
        at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.serialize(TreeTypeAdapter.java:172)
        at com.shade.decima.ui.data.viewer.model.DMFExporter$JsonListSerializer.serialize(DMFExporter.java:1505)
        at com.shade.decima.ui.data.viewer.model.DMFExporter$JsonListSerializer.serialize(DMFExporter.java:1497)
        at com.google.gson.internal.bind.TreeTypeAdapter.write(TreeTypeAdapter.java:88)
        at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.write(TypeAdapterRuntimeTypeWrapper.java:70)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.write(ReflectiveTypeAdapterFactory.java:196)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.write(ReflectiveTypeAdapterFactory.java:366)
        at com.google.gson.Gson.toJson(Gson.java:825)
        at com.shade.decima.ui.data.viewer.model.DMFExporter.export(DMFExporter.java:115)
        at com.shade.decima.ui.data.viewer.model.ModelViewerPanel.export(ModelViewerPanel.java:145)
        at com.shade.decima.ui.data.viewer.model.ModelViewerPanel.lambda$new$1(ModelViewerPanel.java:76)
        ... 7 more
20:29:08.457 [AWT-EventQueue-0] ERROR com.shade.decima.ui.Application - Unhandled exception
java.lang.RuntimeException: java.lang.IndexOutOfBoundsException
        at com.shade.decima.ui.data.viewer.model.ModelViewerPanel.lambda$new$1(ModelViewerPanel.java:79)
        at com.shade.platform.ui.dialogs.ProgressDialog$1.doInBackground(ProgressDialog.java:37)
        at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IndexOutOfBoundsException: null
        at java.base/java.util.zip.DeflaterOutputStream.write(DeflaterOutputStream.java:203)
        at java.base/java.io.ByteArrayInputStream.transferTo(ByteArrayInputStream.java:208)
        at com.shade.decima.ui.data.viewer.model.dmf.DMFInternalBuffer.serialize(DMFInternalBuffer.java:26)
        at com.shade.decima.ui.data.viewer.model.DMFExporter$JsonBufferSerializer.serialize(DMFExporter.java:1515)
        at com.shade.decima.ui.data.viewer.model.DMFExporter$JsonBufferSerializer.serialize(DMFExporter.java:1511)
        at com.google.gson.internal.bind.TreeTypeAdapter.write(TreeTypeAdapter.java:88)
        at com.google.gson.Gson.toJson(Gson.java:825)
        at com.google.gson.Gson.toJsonTree(Gson.java:695)
        at com.google.gson.Gson.toJsonTree(Gson.java:672)
        at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.serialize(TreeTypeAdapter.java:172)
        at com.shade.decima.ui.data.viewer.model.DMFExporter$JsonListSerializer.serialize(DMFExporter.java:1505)
        at com.shade.decima.ui.data.viewer.model.DMFExporter$JsonListSerializer.serialize(DMFExporter.java:1497)
        at com.google.gson.internal.bind.TreeTypeAdapter.write(TreeTypeAdapter.java:88)
        at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.write(TypeAdapterRuntimeTypeWrapper.java:70)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.write(ReflectiveTypeAdapterFactory.java:196)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.write(ReflectiveTypeAdapterFactory.java:366)
        at com.google.gson.Gson.toJson(Gson.java:825)
        at com.shade.decima.ui.data.viewer.model.DMFExporter.export(DMFExporter.java:115)
        at com.shade.decima.ui.data.viewer.model.ModelViewerPanel.export(ModelViewerPanel.java:145)
        at com.shade.decima.ui.data.viewer.model.ModelViewerPanel.lambda$new$1(ModelViewerPanel.java:76)
        ... 7 common frames omitted
```

**Comment by REDxEYE** [comment]
@mithkr Can you please provide file path for file that causes crash?

**Comment by mithkr** [comment]
models/characters/humans/aloy/animation/parts/aloy_ct_a_lx.core
![grafik](https://github.com/ShadelessFox/decima/assets/49768797/c0de01b2-9270-4456-ba39-89ad43feda5a)

**Comment by mithkr** [comment]
Forget to mentioned, that 0.1.12 has an additional issue for which I have  a fix (see diff at the end) and then stumbled over that other issue introduced with 0.1.10:
21:19:25.660 [AWT-EventQueue-0] ERROR com.shade.platform.Launcher - Unhandled exception
java.lang.RuntimeException: java.lang.IllegalArgumentException: Object is not an instance of GGUUID
        at com.shade.decima.ui.data.viewer.model.ModelViewerPanel.lambda$new$1(ModelViewerPanel.java:78)
        at com.shade.platform.ui.dialogs.ProgressDialog$1.doInBackground(ProgressDialog.java:37)
        at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalArgumentException: Object is not an instance of GGUUID
        at com.shade.decima.model.rtti.RTTIUtils.uuidToString(RTTIUtils.java:14)
        at com.shade.decima.ui.data.viewer.model.DMFExporter.exportHZDMeshData(DMFExporter.java:1049)
        at com.shade.decima.ui.data.viewer.model.DMFExporter.regularSkinnedMeshResourceToModel(DMFExporter.java:991)
        at com.shade.decima.ui.data.viewer.model.DMFExporter.toModel(DMFExporter.java:487)
        at com.shade.decima.ui.data.viewer.model.DMFExporter.lodMeshResourceToModel(DMFExporter.java:850)
        at com.shade.decima.ui.data.viewer.model.DMFExporter.exportLodMeshResource(DMFExporter.java:395)
        at com.shade.decima.ui.data.viewer.model.DMFExporter.exportResource(DMFExporter.java:146)
        at com.shade.decima.ui.data.viewer.model.DMFExporter.export(DMFExporter.java:126)
        at com.shade.decima.ui.data.viewer.model.DMFExporter.export(DMFExporter.java:113)
        at com.shade.decima.ui.data.viewer.model.ModelViewerPanel.export(ModelViewerPanel.java:144)
        at com.shade.decima.ui.data.viewer.model.ModelViewerPanel.lambda$new$1(ModelViewerPanel.java:75)
        ... 7 common frames omitted
java.lang.RuntimeException: java.lang.IllegalArgumentException: Object is not an instance of GGUUID
        at com.shade.decima.ui.data.viewer.model.ModelViewerPanel.lambda$new$1(ModelViewerPanel.java:78)
        at com.shade.platform.ui.dialogs.ProgressDialog$1.doInBackground(ProgressDialog.java:37)
        at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalArgumentException: Object is not an instance of GGUUID
        at com.shade.decima.model.rtti.RTTIUtils.uuidToString(RTTIUtils.java:14)
        at com.shade.decima.ui.data.viewer.model.DMFExporter.exportHZDMeshData(DMFExporter.java:1049)
        at com.shade.decima.ui.data.viewer.model.DMFExporter.regularSkinnedMeshResourceToModel(DMFExporter.java:991)
        at com.shade.decima.ui.data.viewer.model.DMFExporter.toModel(DMFExporter.java:487)
        at com.shade.decima.ui.data.viewer.model.DMFExporter.lodMeshResourceToModel(DMFExporter.java:850)
        at com.shade.decima.ui.data.viewer.model.DMFExporter.exportLodMeshResource(DMFExporter.java:395)
        at com.shade.decima.ui.data.viewer.model.DMFExporter.exportResource(DMFExporter.java:146)
        at com.shade.decima.ui.data.viewer.model.DMFExporter.export(DMFExporter.java:126)
        at com.shade.decima.ui.data.viewer.model.DMFExporter.export(DMFExporter.java:113)
        at com.shade.decima.ui.data.viewer.model.ModelViewerPanel.export(ModelViewerPanel.java:144)
        at com.shade.decima.ui.data.viewer.model.ModelViewerPanel.lambda$new$1(ModelViewerPanel.java:75)
        ... 7 more

```
--- a/modules/decima-ui/src/main/java/com/shade/decima/ui/data/viewer/model/DMFExporter.java
+++ b/modules/decima-ui/src/main/java/com/shade/decima/ui/data/viewer/model/DMFExporter.java
@@ -1046,7 +1046,7 @@ public class DMFExporter extends BaseModelExporter implements ModelExporter {
                         }
 
                         final String dataSourceLocation = dataSource.location.substring(dataSource.location.indexOf(":") + 1);
-                        buffer = createDataBuffer(dataSourceLocation + "_" + RTTIUtils.uuidToString(streamObj.obj("Hash")),
+                        buffer = createDataBuffer(dataSourceLocation + "_" + streamObj.obj("Hash").toString(),
                             dataSource, vertexStreamOffset, stride * vertices.vertexCount);
                         resourceLength = dataSource.length;
 
@@ -1092,7 +1092,7 @@ public class DMFExporter extends BaseModelExporter implements ModelExporter {
                 } else {
                     final HZDDataSource dataSource = indices.dataSource.cast();
                     final String dataSourceLocation = dataSource.location.substring(dataSource.location.indexOf(":") + 1);
-                    buffer = createDataBuffer(dataSourceLocation + "_" + RTTIUtils.uuidToString(indices.hash), dataSource, dataSource.getOffset(), indices.getIndexSize() * indices.indexCount);
+                    buffer = createDataBuffer(dataSourceLocation + "_" + indices.hash.toString(), dataSource, dataSource.getOffset(), indices.getIndexSize() * indices.indexCount);
                 }
                 final DMFBufferView bufferView = new DMFBufferView(scene.buffers.indexOf(buffer), 0, indices.getIndexSize() * indices.indexCount);
                 primitive.setIndexBufferView(bufferView, scene);

```

**Comment by ShadelessFox** [comment]
I just fixed this issue in acf47d34502b5346752f8d27d7d324ce52ba0d7f and immediately forgot about it, sorry 😅. Please try building from `master`.

**Comment by mithkr** [comment]
I have now checked out your latest version and it works. I'm wondering, why I run in the IndexOutOfBoundsException after I tried to fix the 0.1.12 issue with the above mentioned patch. I tried than the older release and found that 0.1.10 was the first failed release and the 0.1.9 the last working. Than I started bisecting the changes between 0.1.9 and 0.1.10 and found this commit https://github.com/ShadelessFox/decima/commit/c81ef53989c389154849a847b49d60fc238c3011. If you check out that commit is it working for you?
